crates/router/src/connector/intuit/transformers.rs

```diff
@@ -0,0 +1,244 @@
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    connector::utils,
+    core::errors,
+    pii::PeekInterface,
+    types::{self, api, storage::enums},
+};
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct IntuitPaymentsRequest {
+    amount: String,
+    currency: String,
+    description: String,
+    context: Context,
+    card: Card,
+    capture: bool,
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct Card {
+    number: String,
+    exp_month: String,
+    exp_year: String,
+    cvc: String,
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct Context {
+    mobile: bool,
+    is_ecommerce: bool,
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for IntuitPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        match item.request.payment_method_data {
+            api::PaymentMethod::Card(ref ccard) => {
+                let submit_for_settlement = matches!(
+                    item.request.capture_method,
+                    Some(enums::CaptureMethod::Automatic) | None
+                );
+                Ok(Self {
+                    amount: item.request.amount.to_string(),
+                    currency: item.request.currency.to_string().to_uppercase(),
+                    context: Context {
+                        mobile: item.request.browser_info.clone().map_or(true, |_| false),
+                        is_ecommerce: false,
+                    },
+                    card: Card {
+                        number: ccard.card_number.peek().clone(),
```

Comment:
Don't peek the secret strings, Make the fields in the `IntuitPaymentsRequest ` as secret wherever necessary